### ISSUE TITLE
selecting multiple photos not possible when using SpectrWM

### ISSUE THREAD

**Issue by blitzgneisserin** [darktable-org/darktable#10886]
The problem is this: if using the tiling windows manager SpectrWM, selecting multiple photos (thumbnails in lighttable mode) does not work in dt. ctrl+click has the same result as simple click. In addition, in darktable mode, adjusting the brush with ctrl+scroll and shift+scroll does not work either.

I have tested this in several different window managers and desktop environments such as KDE Plasma or Awesome WM. Ctrl+click (etc.) do work everywhere but SpectrWM. However, in darktable/lighttable mode, ctrl+a for selecting all photos does work.
Ctrl+click does work in other apps such as graphical file managers.

I would also be happy for a workaround, but in the dt shortcut settings, apparently I cannot create a different shortcut because the action "select multiple" does not exist.

This issue is also reported here: https://github.com/conformal/spectrwm/issues/492#issue

OS: Debian 11
Graphics: Nvidia
dt version: 3.9 (dev, updated yesterday)

Thanks in advance for looking into this.

**Comment by johnny-bit** [comment]
Please link it. It's important to keep linked issues linked.

As for multiple selects - i think it doesn't use InputNG but normal key handling, so you could try xinput test to see if same codes are generated under SpectrWM and other WMs or not. And if so - do other GTK apps behave similarly or is the issues isolated to darktable.

**Comment by blitzgneisserin** [comment]
> And if so - do other GTK apps behave similarly or is the issues isolated to darktable.

Apparently the issue only exists in dt - tried Pcmanfm and Thunar, no problems with selecting multiple.

**Comment by blitzgneisserin** [comment]
ok. apparently there is already the alternative shortcut hover+space in place for this

**Comment by dterrahe** [comment]
Is this issue new for 3.8 or did it exist in 3.6.1 as well?

> I cannot create a different shortcut

Just to clarify something here; a "shortcut" normally means a key combination (input ng extends this to key/mouse click/move/scroll combination) that achieves an action without actually having to got to the object that the action is performed on. Whereas here you are looking for a modifier for a mouse action that is performed on the object you are currently pointing at.

Not to say that input ng (and the "shortcut" system) could not be in future extended to encompass center screen mouse actions  as well, but this is currently not the case.

Having said that, input ng _does_ allow you to define alternate modifier keys (which also work in the center view). This is really meant for midi keys (or future other supported input devices), but you could in theory define another keyboard key (not a standard modifier like shift/ctrl/alt, which might be intercepted by the wm) to also act as a ctrl key by mapping it to the _global/modifiers_ action, with element _ctrl_ and effect _hold_.

**Comment by c0dev0id** [comment]
I'm having the same issue with spectrwm + darktable 3.6.1

But I'm happy with the hover+space method. I didn't even know that control+click was supposed to work before reading the issue ;)

**Comment by b0nt4kun** [comment]
Hi, I'm also experiencing this issue with darktable 3.8.1 and spectrwm 3.4.1. Not sure it is relevant but I'm using nvidia-drivers 470.103.01 on a gentoo distribution. <Space bar> + Mouse over do the trick as a workaround for Ctrl+Left Click, but unless I'm mistaken, I did not saw any workaround for Shift+Left Click which used to let me select a range of photos.

For now I did not found any other application (using Gtk, Qt or something else) having this issue. 

I'm was confident since when Shift and Ctrl selection stopped working, so I recompiled darktable 3.6.1: using Shift+Click worked, but not Ctrl+Click... unless I Ctrl+Click on a very small part of the "picture" (see attached: red area does not work, green area works for Ctrl+selection).

I then came back to darktable to try this trick: it does work for both Shift+Click and Ctrl+Click as long as I click inside the green area when performing the selection of the additionnal/last picture:
![2022-02-13_18-30](https://user-images.githubusercontent.com/50054295/153767229-3a3d7467-f036-4e66-98f6-b1d121a5ef4d.png)

Hope this helps, since I don't think it is the intended behaviour.
